src/mongocxx/database.cpp

```diff
@@ -157,5 +165,17 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const database::impl& database::_get_impl() const {
+    if(!_impl) {
+        throw logic_error{make_error_code(error_code::k_invalid_database_object)};
+    }
+    return *_impl;
+}
+
+database::impl& database::_get_impl() {
+    auto cthis = const_cast<const database*>(this);
```

Comment:
is this const_cast-fu worth it? or should we actually just duplicate the implementation?